tests/test_google_analytics_pagination.py

```diff
@@ -2,93 +2,113 @@
+                self.assertGreater(record_count_by_stream.get(stream, -1), self.API_LIMIT,
+                                   msg="We didn't guarantee pagination. The number of records should exceed the api limit.")
 
-#                     # Verify that the automatic fields are sent to the target for paginated streams
-#                     self.assertTrue(auto_fields.issubset(actual_keys),
-#                                     msg="A paginated synced stream has a record that is missing automatic fields.")
+                data = synced_records.get(stream, [])
+                record_messages_keys = [set(row['data'].keys()) for row in data['messages']]
+                auto_fields = self.expected_automatic_fields().get(stream)
 
-#                     # Verify we have more fields sent to the target than just automatic fields (this is set above)
-#                     self.assertEqual(set(), auto_fields.difference(actual_keys),
-#                                      msg="A paginated synced stream has a record that is missing expected fields.")
+                for actual_keys in record_messages_keys:
 
-#                 # TODO Verify by pks that the replicated records match our expectations
-#                 # self.assertPKsEqual(stream, expected_records.get(stream), actual_records)
+                    # Verify that the automatic fields are sent to the target for paginated streams
+                    self.assertTrue(auto_fields.issubset(actual_keys),
+                                    msg="A paginated synced stream has a record that is missing automatic fields.")
+
+                    # Verify we have more fields sent to the target than just automatic fields (this is set above)
+                    self.assertEqual(set(), auto_fields.difference(actual_keys),
+                                     msg="A paginated synced stream has a record that is missing expected fields.")
+
+                
+                messages = synced_records[stream]['messages']
+                
+                # Retrieve records for the start_date window only because, the tap calls the API on day-wise.
+                # So, we are verifying pagination for the start_date API call.
+                start_date_messages = [m for m in messages if m['data']['start_date'] == self.start_date ]
+                primary_keys_list = [tuple([message.get('data').get(expected_pk) for expected_pk in expected_primary_keys])
+                                        for message in start_date_messages
+                                        if message.get('action') == 'upsert']
+
+                primary_keys_list_1 = primary_keys_list[:self.API_LIMIT]
+                primary_keys_list_2 = primary_keys_list[self.API_LIMIT:2*self.API_LIMIT]
+
+                primary_keys_page_1 = set(primary_keys_list_1)
+                primary_keys_page_2 = set(primary_keys_list_2)
+
+                # Verify by primary keys that data is unique for page
```

Comment:
Could you update this validation to ensure all pages are unique rather than the first 2, see example here https://github.com/singer-io/tap-mixpanel/blob/2cf3a86231da042f76f4e3ef9fa8c23600b6f593/tests/tap_tester/test_all_fields_pagination.py#L134-L153

Reply:
Updated pagination test case.